odoo/addons/base/models/ir_attachment.py

```diff
@@ -815,3 +793,18 @@ def _to_http_stream(self):
             stream.size = 0
 
         return stream
+
+    def _can_elevate_access(self, field, access_token=None):
+        attachment_sudo = self.sudo().with_context(prefetch_fields=False)
+        if access_token:
+            if not consteq(attachment_sudo.access_token or "", access_token):
+                raise AccessError("Invalid access token")
+            return True
```

Comment:
I wonder if we really need this raise, it would be fine to just return True when it matches and ignore in other cases.

Reply:
I think if there is an access token and it's not a correct one, we don't want to go any further down for other checks, so either we have to raise here or at least return `False`.